weatherflow2mqtt/weatherflow_mqtt.py

```diff
@@ -166,19 +176,346 @@ async def connect(self) -> None:
+            self.forecast_last_run = now
+
+    def _add_to_queue(
+        self, topic: str, payload: str | None = None, qos: int = 0, retain: bool = False
+    ) -> None:
+        """Add an item to the queue."""
+        self._queue.put_nowait((topic, payload, qos, retain))
+
+    def _device_discovered(self, device: WeatherFlowDevice) -> None:
+        """Handle a discovered device."""
+
+        def _load_complete():
+            print(device, "was loaded")
+            self._setup_sensors(device)
+            device.on(
+                EVENT_STATUS_UPDATE,
+                lambda event: self._handle_status_update_event(device, event),
+            )
+            if isinstance(device, WeatherFlowSensorDevice):
+                device.on(
+                    EVENT_OBSERVATION,
+                    lambda event: self._handle_observation_event(device, event),
+                )
+                if isinstance(device, AirSensorType):
+                    device.on(
+                        EVENT_STRIKE,
+                        lambda event: self._handle_strike_event(device, event),
+                    )
+                if isinstance(device, SkySensorType):
+                    device.on(
+                        EVENT_RAPID_WIND,
+                        lambda event: self._handle_wind_event(device, event),
+                    )
+                    device.on(
+                        EVENT_RAIN_START,
+                        lambda event: self._handle_rain_start_event(device, event),
+                    )
+
+        device.on(EVENT_LOAD_COMPLETE, lambda _: _load_complete())
+
+    def _get_sensor_payload(
+        self,
+        sensor: BaseSensorDescription,
+        serial_number: str,
+        state_topic: str,
+        attr_topic: str,
+        firmware: str,
+    ) -> OrderedDict:
+        """Construct and return a sensor payload."""
+        payload = OrderedDict()
+
+        payload["name"] = f"WF {sensor.name}"
+        payload["unique_id"] = f"{serial_number}-{sensor.id}"
+        if (units := sensor.unit_i if self.is_imperial else sensor.unit_m) is not None:
+            payload["unit_of_measurement"] = units
+        if (device_class := sensor.device_class) is not None:
+            payload["device_class"] = device_class
+        if (state_class := sensor.state_class) is not None:
+            payload["state_class"] = state_class
+        if (icon := sensor.icon) is not None:
+            payload["icon"] = f"mdi:{icon}"
+        payload["state_topic"] = state_topic
+        payload["value_template"] = f"{{{{ value_json.{sensor.id} }}}}"
+        payload["json_attributes_topic"] = attr_topic
+        payload["device"] = {
+            "identifiers": [f"WeatherFlow_{serial_number}"],
+            "connections": [["mac", serial_number]],
+            "manufacturer": "WeatherFlow",
+            "name": "WeatherFlow2MQTT",
+            "model": f"WeatherFlow Weather Station V{VERSION}",
```

Comment:
As above, should we put the actual device model here for name and model? It would help to distinguish the devices under the mqtt integration.

Reply:
Yes, lets do that.